include/sdf/Joint.hh

```diff
@@ -203,12 +203,27 @@ namespace sdf
     /// \param[in] _frame The name of the pose relative-to frame.
     public: void SetPoseRelativeTo(const std::string &_frame);
 
-    /// \brief Get the thread pitch (only valid for screw joints)
-    /// \return The thread pitch
+    /// \brief Get the displacement along the joint axis for each complete
+    /// revolution around the joint axis (only valid for screw joints).
+    /// \return The thread pitch with units of meters per revolution with a
+    /// positive value corresponding to a right-handed thread.
+    public: double ScrewThreadPitch() const;
+
+    /// \brief Set the thread pitch (only valid for screw joints).
+    /// \param[in] _threadPitch The thread pitch with units of meters per
+    /// revolution with a positive value corresponding to a right-handed thread.
+    public: void SetScrewThreadPitch(double _threadPitch);
+
+    /// \brief Get the thread pitch in gazebo-classic format (only valid for
+    /// screw joints). This will be deprecated in a future version.
+    /// \return The thread pitch with units of radians / meters and a positive
+    /// value coresponding to a left-handed thread.
```

Comment:
The documentation [gazebo-classic](https://github.com/gazebosim/gazebo-classic/blob/33f7e47c982a11094203ee7c8addd25598de3e7c/gazebo/physics/ScrewJoint.hh#L64-L68) says it uses right-handed threads. Is that incorrect?

Reply:
yes, I believe the gazebo-classic documentation is incorrect based on the following test expectations:

https://github.com/gazebosim/gazebo-classic/blob/gazebo11/test/integration/joint_screw.cc#L213-L214

~~~
  EXPECT_NEAR(joint_00->Position(0), 0.25*IGN_PI, g_tolerance);
  EXPECT_NEAR(joint_00->Position(1), -0.25*IGN_PI/pitch_00, g_tolerance);
~~~

since the `ScrewJoint::Position(unsigned int _axisIndex)` API [returns the screw angle for `_axisIndex == 0` and the linear screw position for `_axisIndex == 1`](https://github.com/gazebosim/gazebo-classic/blob/gazebo11/gazebo/physics/ode/ODEScrewJoint.cc#L143-L153), that test expectation implies that `linear_screw_position == - screw_angle / thread_pitch`

Reply:
you can also insert the `Demo Joint Types` model into gazebo classic and watch the motion of the screw joint, noting that it has positive thread pitch:

https://github.com/osrf/gazebo_models/blob/master/demo_joint_types/model.sdf#L734

Reply:
Okay, thanks for the clarification.